lightning/connection_common.go

```diff
@@ -0,0 +1,18 @@
+package lightning
+
+import (
+	"context"
+	"io"
+)
+
+// ClnConnection struct
+type ClnConnection struct {
+}
+
+// ClnConnectionAPI interface
+type ClnConnectionAPI interface {
+	// Call calls serviceMethod with args and fills reply with response
+	Call(ctx context.Context, serviceMethod string, args any, reply any) error
+	// StreamResponse is meant for streaming responses it calls serviceMethod with arggs and returns an io.Reader
+	StreamResponse(ctx context.Context, serviceMethod string, args any) (io.Reader, error)
```

Comment:
would make sense to unify naming `call` and `streamResponse`. If i understand correctly both are getting response  except one is streaming it.

Reply:
For `commando` implementations are a bit different, for `unix socket` `StreamResponse` is just a wrapper around `Call` (but that shouldn't be an issue since reading from local socket should be fast, however you still allocate a lot of memory). I didn't want to add another wrapper and/or make this even less comprehensible.